### DISCUSSION TITLE
Start and End Events for `onColumnResized`

### DISCUSSION THREAD

**Discussion by BrianHung** [glideapps/glide-data-grid#254] [Ideas]
A lot of draggable libraries allow for props to listen to the start and end of a drag event at `onMouseDown` and `onMouseUp` respectively.

E.g. https://github.com/react-grid-layout/react-draggable#draggablecore-api https://github.com/react-grid-layout/react-draggable/blob/31798e920647f40308a144a9f989c771755a21db/lib/DraggableCore.js#L424-L434

https://reactflow.dev/docs/api/react-flow-props/#nodes

It would be great if `glide-data-grid` could support this too via `onColumnResizeStart` and `onColumnResizeStop`.

Main reason is using React state to update column widths during dragging state, and then saving that React state into a persistent state on stop so we're not saving each incremental change in width or debouncing it until completion (while also allowing UI to update).

**Comment by jassmith** [comment]
Great idea will add. Maybe onColumnResize should just have an arg saying that it is a "finalized" event. This would require a double emit at the end but keeps the API simpler if you ask me.

**Reply by BrianHung** [reply]
With a "finalized" arg for end, there would also need to be a "start" arg for start (?).

Looking at React events / browser event listener naming conventions, I came up with
https://github.com/glideapps/glide-data-grid/pull/257

**Comment by jassmith** [comment]
These are shipped in 4.0.3
